ballista/scheduler/src/state/task_manager.rs

```diff
@@ -731,6 +737,30 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> TaskManager<T, U>
         }
     }
 
+    pub async fn set_partition_as_replicated(
+        &self,
+        job_id: &str,
+        path: &str,
+    ) -> Result<()> {
+        if let Some(JobStatus {
+            status:
+                Some(job_status::Status::Successful(SuccessfulJob {
+                    mut partition_location,
+                    ..
+                })),
+            ..
+        }) = self.state.get_job_status(job_id).await?
+        {
+            {
+                if let Some(p) = partition_location.iter_mut().find(|p| p.path == path) {
+                    p.replicated = true;
+                }
+            }
```

Comment:
This won't actually update in the state though right? The status you get back from `get_job_status` is a clone, not a mutable reference to the status in the state

Reply:
Removed